src/features/Policy/PolicyModal/index.js

```diff
@@ -1,23 +1,32 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { Flex, Input, Modal, Form, Button } from 'antd';
 import ReactQuill from 'react-quill';
 import 'react-quill/dist/quill.snow.css'; // Import Quill styles
-import { useDispatch } from 'react-redux';
-import { createPolicyApi } from 'src/store/slices/policySlice/apis';
+import { useDispatch, useSelector } from 'react-redux';
+import { createPolicyApi, updatePolicyApi } from 'src/store/slices/policySlice/apis';
+import { getSelectedPolicy } from 'src/store/slices/policySlice/selectors';
 
-const PolicyModal = ({ open, handleClose }) => {
-    const [form] = Form.useForm(); // Create form instance
+const PolicyModal = ({ open, handleClose, selectedPolicy }) => {
+    const [form] = Form.useForm();
 
     const dispatch = useDispatch()
+
+    console.log(selectedPolicy, "selecteddddd")
+
     const handleCancel = () => {
-        form.resetFields(); // Reset fields on cancel
+        form.resetFields();
         handleClose(false);
+
     };
 
     const handleFinish = (values) => {
-        console.log('Form values:', values);
-        dispatch(createPolicyApi(values))
-        handleClose(false); // Close modal after submission
+        if (selectedPolicy) {
+            dispatch(updatePolicyApi({ id: selectedPolicy?._id, data: values }));
+        } else {
+            dispatch(createPolicyApi(values));
+        }
+        form.resetFields();
```

Comment:
create  try catch after  successfully created  or updated then close modal and reset